### PR TITLE
Add make rules for running a subset of e2e tests in a presbumit nomo/multi job

### PR DESCRIPTION
**Pull request by haiyanmeng**
This PR should be submitted after https://github.com/GoogleCloudPlatform/oss-test-infra/pull/1751.

### CODE REVIEW

File: Makefile.e2e [hunk 1]
```diff
@@ -168,12 +168,45 @@ test-e2e-go-ephemeral: __push-local-image
 	# The .sock volume allows you to connect to the Docker daemon of the host.
 	# Part of the docker-in-docker pattern.
 	@docker run -v /var/run/docker.sock:/var/run/docker.sock --network="host" prow-image make __test-e2e-go-nobuild-prow
+# This target runs all the e2e tests with the mono-repo mode.
 test-e2e-go-ephemeral-mono-repo: __push-local-image
 	kind delete clusters --all
 	./scripts/e2e-go-ephemeral.sh --timeout 60m --parallel 18 --image-tag=$(LATEST_IMAGE_TAG)
+# This target runs the first group of e2e tests with the mono-repo mode.
+test-e2e-go-ephemeral-mono-repo-group1: __push-local-image
```
Comment (nan-yu) [inline Makefile.e2e:176]: For consistency, can we rename it to `test-e2e-kind-mono-repo-test-group1`?

We're going to port all remaining bats test to go, so `go` is not needed in the target name.

`kind` indicates it is an ephemeral cluster.

**Comment by haiyanmeng** [comment]
This PR should submitted after https://github.com/GoogleCloudPlatform/oss-test-infra/pull/1751.

File: Makefile.e2e [hunk 2]
```diff
@@ -168,12 +168,45 @@ test-e2e-go-ephemeral: __push-local-image
 	# The .sock volume allows you to connect to the Docker daemon of the host.
 	# Part of the docker-in-docker pattern.
 	@docker run -v /var/run/docker.sock:/var/run/docker.sock --network="host" prow-image make __test-e2e-go-nobuild-prow
+# This target runs all the e2e tests with the mono-repo mode.
```
[the remark this answers is not in the document]
Comment (haiyanmeng) [reply Makefile.e2e:172]: Done.

**Comment by sdowell** [comment]
Consider making this a non-breaking change by having the old targets invoke each of the new (test group) targets. That way we can submit this before the prowjob update and then submit the prowjob update with no breakage. The old targets could then be removed after.

File: Makefile.e2e [hunk 3]
```diff
@@ -163,17 +163,57 @@ __test-e2e-go-nobuild-prow:
...
 	@docker build . -f build/test-e2e-go/Dockerfile -t prow-image
 	# The .sock volume allows you to connect to the Docker daemon of the host.
 	# Part of the docker-in-docker pattern.
 	@docker run -v /var/run/docker.sock:/var/run/docker.sock --network="host" prow-image make __test-e2e-go-nobuild-prow
-test-e2e-go-ephemeral-mono-repo: __push-local-image
+
+# This target runs all the e2e tests with the mono-repo mode.
+test-e2e-kind-mono-repo: __push-local-image
+	kind delete clusters --all
+	./scripts/e2e-kind.sh --timeout 60m --parallel 18 --image-tag=$(LATEST_IMAGE_TAG)
```
[the remark this answers is not in the document]
Comment (haiyanmeng) [reply Makefile.e2e:175]: I don't think we need to do this. Eventually, we will remove `test-e2e-kind-mono-repo` and just keep `test-e2e-kind-mono-repo-test-groupXX`.

[the remark this answers is not in the document]
Comment (nan-yu) [reply Makefile.e2e:175]: Why not keep both? `test-e2e-kind-mono-repo` can be used as a base target, and `test-groupXX` is kind of parameterized for convenience.

[the remark this answers is not in the document]
Comment (haiyanmeng) [reply Makefile.e2e:175]: I can image someone may want to run several e2e tests locally with a kind cluster following https://g3doc.corp.google.com/cloud/kubernetes/g3doc/subgroup/catalyst/config-sync/devguides/dev_guide_cs.md?cl=head#testing-locally. 
I don't really know whether it is common for people to run `make test-e2e-kind-mono-repo` locally.

[the remark this answers is not in the document]
Comment (haiyanmeng) [reply Makefile.e2e:175]: Keeping `test-e2e-kind-mono-repo` as a base target sounds good.

[the remark this answers is not in the document]
Comment (haiyanmeng) [reply Makefile.e2e]: Done. Thanks for the suggestion.

File: Makefile.e2e [hunk 4]
```diff
@@ -163,17 +163,47 @@ __test-e2e-go-nobuild-prow:
...
+
+# This target runs the first group of e2e tests with the multi-repo mode.
+test-e2e-kind-multi-repo-test-group1:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --test-features=acm-controller,cluster-selector,drift-control,lifecycle,nomos-cli" test-e2e-kind-multi-repo
+
+# This target runs the second group of e2e tests with the multi-repo mode.
+test-e2e-kind-multi-repo-test-group2:
+	$(MAKE) E2E_ARGS="$(E2E_ARGS) --test-features=sync-source,reconciliation1" test-e2e-kind-multi-repo
+
+		--test-features=sync-source,reconciliation1"
```
Comment (nan-yu) [inline Makefile.e2e:202]: Delete this?

**Review by nan-yu — APPROVED** [review]
/lgtm

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**